From a18e3afd64b7d6f78b66fea4e8ae492365f5d75d Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 13 Sep 2017 17:46:26 +0800
Subject: [PATCH 4532/5242] MLK-19413-21 drm/imx: dpu: crtc: Peek auxiliary
 CRTC resources

commit  6336ced7ba29f0ffbd2df0b219d7fbb98269e4e6 from
https://source.codeaurora.org/external/imx/linux-imx.git

Cache the auxiliary CRTC resources in struct dpu_crtc via the
dpu_aux_{unit}_peek() helpers so that the DPU CRTC driver may
use them later.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-crtc.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-crtc.c b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
index bd6ccbf..7b5f58b 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-crtc.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
@@ -39,6 +39,11 @@ struct dpu_crtc {
 	struct dpu_extdst	*ed;
 	struct dpu_framegen	*fg;
 	struct dpu_tcon		*tcon;
+	struct dpu_constframe	*aux_cf;
+	struct dpu_disengcfg	*aux_dec;
+	struct dpu_extdst	*aux_ed;
+	struct dpu_framegen	*aux_fg;
+	struct dpu_tcon		*aux_tcon;
 	struct dpu_plane	**plane;
 	unsigned int		hw_plane_num;
 	unsigned int		stream_id;
@@ -587,30 +592,35 @@ static int dpu_crtc_get_resources(struct dpu_crtc *dpu_crtc)
 		ret = PTR_ERR(dpu_crtc->cf);
 		goto err_out;
 	}
+	dpu_crtc->aux_cf = dpu_aux_cf_peek(dpu_crtc->cf);
 
 	dpu_crtc->dec = dpu_dec_get(dpu, stream_id);
 	if (IS_ERR(dpu_crtc->dec)) {
 		ret = PTR_ERR(dpu_crtc->dec);
 		goto err_out;
 	}
+	dpu_crtc->aux_dec = dpu_aux_dec_peek(dpu_crtc->dec);
 
 	dpu_crtc->ed = dpu_ed_get(dpu, stream_id + 4);
 	if (IS_ERR(dpu_crtc->ed)) {
 		ret = PTR_ERR(dpu_crtc->ed);
 		goto err_out;
 	}
+	dpu_crtc->aux_ed = dpu_aux_ed_peek(dpu_crtc->ed);
 
 	dpu_crtc->fg = dpu_fg_get(dpu, stream_id);
 	if (IS_ERR(dpu_crtc->fg)) {
 		ret = PTR_ERR(dpu_crtc->fg);
 		goto err_out;
 	}
+	dpu_crtc->aux_fg = dpu_aux_fg_peek(dpu_crtc->fg);
 
 	dpu_crtc->tcon = dpu_tcon_get(dpu, stream_id);
 	if (IS_ERR(dpu_crtc->tcon)) {
 		ret = PTR_ERR(dpu_crtc->tcon);
 		goto err_out;
 	}
+	dpu_crtc->aux_tcon = dpu_aux_tcon_peek(dpu_crtc->tcon);
 
 	return 0;
 err_out:
-- 
1.7.9.5

